diff --git a/hypervisor/arch/x86/guest/vmsr.c b/hypervisor/arch/x86/guest/vmsr.c
index 65ba1059f..3057da4fa 100644
--- a/hypervisor/arch/x86/guest/vmsr.c
+++ b/hypervisor/arch/x86/guest/vmsr.c
@@ -385,7 +385,7 @@ static void prepare_auto_msr_area(struct acrn_vcpu *vcpu)
                vcpu->arch.msr_area.count++;
        }

-       if (is_platform_rdt_capable()) {
+       if (is_platform_rdt_capable() && is_vcat_configured(vcpu->vm)) {
                struct acrn_vm_config *cfg = get_vm_config(vcpu->vm->vm_id);
                uint16_t vcpu_clos;

@@ -396,7 +396,7 @@ static void prepare_auto_msr_area(struct acrn_vcpu *vcpu)
                /* RDT: only load/restore MSR_IA32_PQR_ASSOC when hv and guest have different settings
                 * vCAT: always load/restore MSR_IA32_PQR_ASSOC
                 */
-               if (is_vcat_configured(vcpu->vm) || (vcpu_clos != hv_clos)) {
+               if (vcpu_clos != hv_clos) {
                        vcpu->arch.msr_area.guest[vcpu->arch.msr_area.count].msr_index = MSR_IA32_PQR_ASSOC;
                        vcpu->arch.msr_area.guest[vcpu->arch.msr_area.count].value = clos2pqr_msr(vcpu_clos);
                        vcpu->arch.msr_area.host[vcpu->arch.msr_area.count].msr_index = MSR_IA32_PQR_ASSOC;

