diff --git a/misc/config_tools/schema/config.xsd b/misc/config_tools/schema/config.xsd
index d427d4743..065f90a78 100644
--- a/misc/config_tools/schema/config.xsd
+++ b/misc/config_tools/schema/config.xsd
@@ -139,7 +139,7 @@ If your VM is not a security VM, leave this option unchecked. </xs:documentation
         </xs:restriction>
       </xs:simpleType>
     </xs:element>
-    <xs:element name="VUART_TX_BUF_SIZE" default="8192">
+    <xs:element name="VUART_TX_BUF_SIZE" default="81920">
       <xs:annotation acrn:title="vuart tx buffer size" acrn:views="advanced"
                      acrn:errormsg="'required': 'must config the max tx buffer size of vuart in byte'">
         <xs:documentation>Specify the maximum tx buffer size of vuart.</xs:documentation>

