diff --git a/hypervisor/arch/x86/guest/vmcs.c b/hypervisor/arch/x86/guest/vmcs.c
index 689e4a846..b90254722 100644
--- a/hypervisor/arch/x86/guest/vmcs.c
+++ b/hypervisor/arch/x86/guest/vmcs.c
@@ -581,6 +581,9 @@ void init_vmcs(struct acrn_vcpu *vcpu)
        init_guest_state(vcpu);
        init_entry_ctrl(vcpu);
        init_exit_ctrl(vcpu);
+
+       if (is_x2apic_enabled(&vcpu->arch.vlapic))
+               switch_apicv_mode_x2apic(vcpu);
 }

 /**

